assets/pages/new-contributions-landing/components/ContributionAmount.jsx

```diff
@@ -68,25 +70,27 @@ const renderAmount = (currency: Currency, spokenCurrency: SpokenCurrency, props:
 
 
 function ContributionAmount(props: PropTypes) {
+  console.dir(props.selectedAmounts);
+  const validAmounts: Amount[] = amounts('notintest')[props.contributionType][props.countryGroupId];
   return (
     <fieldset className={classNameWithModifiers('form__radio-group', ['pills', 'contribution-amount'])}>
       <legend className={classNameWithModifiers('form__legend', ['radio-group'])}>Amount</legend>
       <ul className="form__radio-group-list">
-        {amounts('notintest')[props.contributionType][props.countryGroupId].map(renderAmount(currencies[props.currency], spokenCurrencies[props.currency], props))}
+        {validAmounts.map(renderAmount(currencies[props.currency], spokenCurrencies[props.currency], props))}
         <li className="form__radio-group-item">
           <input
             id="contributionAmount-other"
             className="form__radio-group-input"
             type="radio"
             name="contributionAmount"
             value="other"
-            checked={props.showOther}
-            onChange={props.selectOtherAmount}
+            checked={props.selectedAmounts[props.contributionType] === validAmounts.length + 1}
+            onChange={props.selectOtherAmount(props.contributionType, validAmounts.length + 1)}
           />
           <label htmlFor="contributionAmount-other" className="form__radio-group-label">Other</label>
         </li>
       </ul>
-      {props.showOther ? (
+      {props.selectedAmounts[props.contributionType] === validAmounts.length + 1 ? (
```

Comment:
It's not immediately apparent to me what this condition is checking (whereas `props.showOther` was very clear). Could we pull it out into a prop or a helper function? Especially since we're checking it in at least two places